accession-commons-monotonic-generator-jpa/src/main/java/uk/ac/ebi/ampt2d/commons/accession/persistence/jpa/monotonic/service/ContiguousIdBlockService.java

```diff
@@ -49,14 +54,19 @@ public void save(Iterable<ContiguousIdBlock> blocks) {
     public ContiguousIdBlock reserveNewBlock(String categoryId, String instanceId) {
         ContiguousIdBlock lastBlock = repository.findFirstByCategoryIdOrderByLastValueDesc(categoryId);
         BlockParameters blockParameters = getBlockParameters(categoryId);
+        ContiguousIdBlock reservedBlock;
         if (lastBlock != null) {
-            return repository.save(lastBlock.nextBlock(instanceId, blockParameters.getBlockSize(),
-                    blockParameters.getNextBlockInterval(), blockParameters.getBlockStartValue()));
+            reservedBlock = repository.save(lastBlock.nextBlock(instanceId, blockParameters.getBlockSize(),
+                                                                blockParameters.getNextBlockInterval(),
+                                                                blockParameters.getBlockStartValue()));
         } else {
             ContiguousIdBlock newBlock = new ContiguousIdBlock(categoryId, instanceId,
-                    blockParameters.getBlockStartValue(), blockParameters.getBlockSize());
-            return repository.save(newBlock);
+                                                               blockParameters.getBlockStartValue(),
+                                                               blockParameters.getBlockSize());
+            reservedBlock = repository.save(newBlock);
         }
+        entityManager.flush();
```

Comment:
I understand that this might be a herculean task but did you check if it is possible to write a test that fails without the flush statement here but succeeds with it?

Reply:
In the scope of this PR to honor database constraint violations, we can potentially succeed on the Postgres side by potentially flushing reserved blocks. But on the Mongo side, we still have the insidious [save-watermark-after-mongo-save](https://github.com/EBIvariation/accession-commons/blob/5aaa16936bbe7dcd877ccc323b1d144b9836c6e2/accession-commons-core/src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicAccessioningService.java#L93) scenario which can lead to overlapping allocations **even when we run accession for multiple species sequentially with the same instance ID**. 

Scenario:

**Step 1:** species 1 -> instance-1 -> block allocation requested (100-200) -> mongo save for 10 records (partial failure for some reason with 5 written) -xx-> watermark NOT updated for the 5 saved records in the block

**Step 2:** We fail to check results carefully for the above species (although unlikely if we chain the clustering executions in Nextflow) and proceed with Species 2

species 2 -> instance-1 -> block allocation (100-200) -> mongo save for 10 records (succeeds with possible overlap of 5 accessions with species 1) -> watermark updated for 10 saved accessions in the blcok